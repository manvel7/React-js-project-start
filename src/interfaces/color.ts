interface ITextPalette {
  dark: string
  darkDimmed: string
  white: string
}

interface IBackgroundPalette {
  dark: string
  white: string
  light: string
  disabled: string
  tinted: string
}

interface IAlertPalette {
  info: string
  success: string
  warn: string
  error: string
}

interface IPalette {
  primary: string
  text: ITextPalette
  background: IBackgroundPalette
  border: string
  disabled: string
  alert: IAlertPalette
}

export {IPalette}
