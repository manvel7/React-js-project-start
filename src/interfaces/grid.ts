type ColBreakpointName = "xs" | "sm" | "md" | "lg" | "xl"
type ScreenBreakpointName = ColBreakpointName | "xxl"

type ColBreakpoints = Record<ColBreakpointName, string>
type GridBreakpoints = Record<ScreenBreakpointName, string>
type ContainerMaxWidths = Partial<Record<ScreenBreakpointName, string>>

interface IGrid {
  columns: number
  gutterWidth: string
  breakpoints: GridBreakpoints
  containerMaxWidths: ContainerMaxWidths
}

export {
  ColBreakpointName,
  ColBreakpoints,
  ScreenBreakpointName,
  ContainerMaxWidths,
  GridBreakpoints,
  IGrid
}
