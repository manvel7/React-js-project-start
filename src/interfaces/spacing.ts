interface ISpacing {
  default: string
  defaultX: string
  defaultY: string
}

export {ISpacing}
