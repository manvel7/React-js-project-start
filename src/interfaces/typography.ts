interface ITypographyStyles {
  size: string
  weight: number
  lineHeight: string
  letterSpacing: string
}

type TypographyVariantName =
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "body"
  | "caption"

type TypographyVariants = Record<TypographyVariantName, ITypographyStyles>

type FontWeightName = "normal" | "bold" | "boldest"

type FontWeights = Record<FontWeightName, number>

interface ITypography {
  htmlFontSize: string
  fontFamily: string
  fontWeights: FontWeights
  variants: TypographyVariants
}

export {ITypography}
