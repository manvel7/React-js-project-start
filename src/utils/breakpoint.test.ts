import {GridBreakpoints} from "../interfaces/grid"
import {
  mediaBreakpointBetween,
  mediaBreakpointDown,
  mediaBreakpointOnly,
  mediaBreakpointUp
} from "./breakpoint"

describe("utils / breakpoint", () => {
  const breakpoints: GridBreakpoints = {
    xs: "0px",
    sm: "576px",
    md: "768px",
    lg: "992px",
    xl: "1200px",
    xxl: "1456px"
  }

  describe("mediaBreakpointUp", () => {
    it("should match snapshot", () => {
      const mediaUpStyles = mediaBreakpointUp(
        "sm",
        breakpoints,
        "color: black;"
      )

      expect(mediaUpStyles).toMatchSnapshot()
    })
  })

  describe("mediaBreakpointDown", () => {
    it("should match snapshot", () => {
      const mediaDownStyles = mediaBreakpointDown(
        "md",
        breakpoints,
        "color: black;"
      )

      expect(mediaDownStyles).toMatchSnapshot()
    })
  })

  describe("mediaBreakpointBetween", () => {
    it("should match snapshot", () => {
      const mediaBetweenStyles = mediaBreakpointBetween(
        "sm",
        "lg",
        breakpoints,
        "color: black;"
      )

      expect(mediaBetweenStyles).toMatchSnapshot()
    })
  })

  describe("mediaBreakpointOnly", () => {
    it("should match snapshot", () => {
      const mediaOnlyStyles = mediaBreakpointOnly(
        "md",
        breakpoints,
        "color: black;"
      )

      expect(mediaOnlyStyles).toMatchSnapshot()
    })
  })
})
