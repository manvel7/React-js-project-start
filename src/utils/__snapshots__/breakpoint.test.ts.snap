// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils / breakpoint mediaBreakpointBetween should match snapshot 1`] = `
"
      @media (min-width: 576px) and (max-width: 1199px) {
        color: black;
      }
    "
`;

exports[`utils / breakpoint mediaBreakpointDown should match snapshot 1`] = `
"
      @media (max-width: 991px) {
        color: black;
      }
    "
`;

exports[`utils / breakpoint mediaBreakpointOnly should match snapshot 1`] = `
"
      @media (min-width: 768px) and (max-width: 991px) {
        color: black;
      }
    "
`;

exports[`utils / breakpoint mediaBreakpointUp should match snapshot 1`] = `
"
      @media (min-width: 576px) {
        color: black;
      }
    "
`;
